kube/src/api/admission.rs

```diff
@@ -0,0 +1,223 @@
+    /// the caller provided `meta.k8s.io/v1.PatchOptions`.
+    pub options: Option<RawExtension>,
+}
+
+/// The operation specified in an [`AdmissionRequest`].
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+pub enum Operation {
+    /// An operation that creates a resource.
+    CREATE,
+    /// An operation that updates a resource.
+    UPDATE,
+    /// An operation that deletes a resource.
+    DELETE,
+    /// An operation that connects to a resource.
+    CONNECT,
+}
+
+/// An outgoing [`AdmissionReview`] response.
+#[derive(Serialize, Deserialize, Clone, Debug, Default, PartialEq)]
+#[serde(rename_all = "camelCase")]
+pub struct AdmissionResponse {
+    /// Identifier for the individual request/response. This must be copied over
+    /// from the corresponding AdmissionRequest.
+    pub uid: String,
+    /// Indicates whether or not the admission request was permitted.
+    pub allowed: bool,
+    /// Extra details into why an admission request was denied. This field IS NOT
+    /// consulted in any way if "Allowed" is "true".
+    pub result: Status,
+    /// The patch body. Currently we only support "JSONPatch" which implements
+    /// RFC 6902.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub patch: Option<Vec<u8>>,
+    /// The type of Patch. Currently we only allow "JSONPatch".
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub patch_type: Option<PatchType>,
+    /// An unstructured key value map set by remote admission controller (e.g.
+    /// error=image-blacklisted). MutatingAdmissionWebhook and
+    /// ValidatingAdmissionWebhook admission controller will prefix the keys with
+    /// admission webhook name (e.g.
+    /// imagepolicy.example.com/error=image-blacklisted). AuditAnnotations will
+    /// be provided by the admission webhook to add additional context to the
+    /// audit log for this request.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub audit_annotations: Option<HashMap<String, String>>,
+    /// A list of warning messages to return to the requesting API client.
+    /// Warning messages describe a problem the client making the API request
+    /// should correct or be aware of. Limit warnings to 120 characters if
+    /// possible. Warnings over 256 characters and large numbers of warnings may
+    /// be truncated.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub warnings: Option<Vec<String>>,
+}
+
+impl AdmissionResponse {
+    /// Converts an [`AdmissionResponse`] into a generic [`AdmissionReview`] that
+    /// can be used as a webhook response.
+    pub fn into_review(self) -> AdmissionReview<DynamicObject> {
+        AdmissionReview {
+            typ: TypeMeta {
+                kind: "AdmissionReview".to_owned(),
+                api_version: "admission.k8s.io/v1".to_owned(),
+            },
+            response: Some(self),
+            ..Default::default()
```

Comment:
This is probably what causes the default. I think we can just set `request: None` and remove `Default` requirements.